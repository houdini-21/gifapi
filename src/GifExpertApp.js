import React, { useState } from 'react';
import AddCatergories from './components/AddCatergories';
import GifGrid from './components/GifGrid';
import BackgroundParticles from './components/BackgroundParticles';

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  /*
     *
  const addCategory = () => {
    setCategories([...categories, 'eyeshield 21']);
  };
        * */

  return (
    <div className="principal__body">
      <BackgroundParticles />
      <div className="principal__header">
        <h2 className="title">GifExpertApp</h2>
        <AddCatergories setCategories={setCategories} />
      </div>
      {categories.map((item, indexArray) => (
        <div className="box__items-category" key={indexArray}>
          <h3 className="title__category">{item}</h3>
          <GifGrid category={item} />
        </div>
      ))}
    </div>
  );
};

export default GifExpertApp;
