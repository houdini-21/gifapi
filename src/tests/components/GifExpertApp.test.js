import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../../GifExpertApp';

describe('pruebas en GifExpertApp', () => {
  test('haciendo snapshot del componente', () => {
    let wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar una lista de categorias', () => {
    const categories = ['silicon valley'];

    let wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
