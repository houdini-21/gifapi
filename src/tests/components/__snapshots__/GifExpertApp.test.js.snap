// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifExpertApp debe de mostrar una lista de categorias 1`] = `
<div
  className="principal__body"
>
  <BackgroundParticles />
  <div
    className="principal__header"
  >
    <h2
      className="title"
    >
      GifExpertApp
    </h2>
    <AddCatergories
      setCategories={[Function]}
    />
  </div>
  <div
    className="box__items-category"
    key="0"
  >
    <h3
      className="title__category"
    >
      silicon valley
    </h3>
    <GifGrid
      category="silicon valley"
    />
  </div>
</div>
`;

exports[`pruebas en GifExpertApp haciendo snapshot del componente 1`] = `
<div
  className="principal__body"
>
  <BackgroundParticles />
  <div
    className="principal__header"
  >
    <h2
      className="title"
    >
      GifExpertApp
    </h2>
    <AddCatergories
      setCategories={[Function]}
    />
  </div>
</div>
`;
